# Exploratory data analysis

# Overview 

This project will utilize movie data from various sources to create recommendations for a company to take to create a movie studio. It will draw findings from analysis to determine the best metrics to focus on for the recommendations.

In [96]:
# import needed libraries
import pandas as pd
import sqlite3

# Data Sources

- Box Office Mojo
- Rotten Tomatoes
- The Movie Database
- The Numbers 
- IMDB

Let's go through each dataset and see what we have.

## BOX OFFICE MOJO

In [90]:
bom_movie_gross_df = pd.read_csv('/Users/nechamaborisute/Desktop/phase-2-project/zippedData/bom.movie_gross.csv.gz', compression  = 'gzip')

In [3]:
bom_movie_gross_df.head()

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   
3                                    Inception     WB     292600000.0   
4                          Shrek Forever After   P/DW     238700000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010  
3     535700000  2010  
4     513900000  2010

In [4]:
bom_movie_gross_df.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


In [92]:
bom_movie_gross_df.describe()

domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

There are 3387 records in this dataframe and each represent a movie with title, studio, domestic gross, foriegn gross, and year it was released.

columns: 
- title: movie title 
- studio: name of the studio
- domestic gross: money made domestically
- foreign gross: money made foreignly 
    - third are nulls
- year: year movie was released



In [5]:
bom_movie_gross_df.isna().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

Going to drop the nulls in all columns because there aren't too many, with the exception of foriegn gross but ill drop them for now and see what info the other datasets have.

In [6]:
bom_movie_gross_df = bom_movie_gross_df.dropna()

In [7]:
bom_movie_gross_df.isna().sum()

title             0
studio            0
domestic_gross    0
foreign_gross     0
year              0
dtype: int64

In [8]:
bom_movie_gross_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 2007 entries, 0 to 3353
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           2007 non-null   object 
 1   studio          2007 non-null   object 
 2   domestic_gross  2007 non-null   float64
 3   foreign_gross   2007 non-null   object 
 4   year            2007 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 94.1+ KB


This clean data set consists of 2007 records.

## ROTTEN TOMATOES

In [13]:
rt_movie_info = pd.read_csv('/Users/nechamaborisute/Desktop/phase-2-project/zippedData/rt.movie_info.tsv.gz', compression = 'gzip', sep = '\t')

In [14]:
rt_movie_info.head()

id                                           synopsis rating  \
0   1  This gritty, fast-paced, and innovative police...      R   
1   3  New York City, not-too-distant-future: Eric Pa...      R   
2   5  Illeana Douglas delivers a superb performance ...      R   
3   6  Michael Douglas runs afoul of a treacherous su...      R   
4   7                                                NaN     NR   

                                 genre          director  \
0  Action and Adventure|Classics|Drama  William Friedkin   
1    Drama|Science Fiction and Fantasy  David Cronenberg   
2    Drama|Musical and Performing Arts    Allison Anders   
3           Drama|Mystery and Suspense    Barry Levinson   
4                        Drama|Romance    Rodney Bennett   

                            writer  theater_date      dvd_date currency  \
0                   Ernest Tidyman   Oct 9, 1971  Sep 25, 2001      NaN   
1     David Cronenberg|Don DeLillo  Aug 17, 2012   Jan 1, 2013        $   
2                   Allison Anders  Sep 13, 1996  Apr 18, 2000      NaN   
3  Paul Attanasio|Michael Crichton   Dec 9, 1994  Aug 27, 1997      NaN   
4                     Giles Cooper           NaN           NaN      NaN   

  box_office      runtime             studio  
0        NaN  104 minutes                NaN  
1    600,000  108 minutes  Entertainment One  
2        NaN  116 minutes                NaN  
3        NaN  128 minutes                NaN  
4        NaN  200 minutes                NaN

In [15]:
rt_movie_info.shape

(1560, 12)

In [16]:
rt_movie_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


The Rotten Tomatoes movie info dataset includes 1560 records where each record represents a movie, with id, rating, director, writer, box office, and studio as some of the columns.

There seem to be lots of nulls in important columns such as studio and box office so may not use this dataset.

In [17]:
rt_reviews = pd.read_csv('/Users/nechamaborisute/Desktop/phase-2-project/zippedData/rt.reviews.tsv.gz', compression = 'gzip', sep = '\t', encoding = 'latin') # encoding is utf8?

In [18]:
rt_reviews.head()

id                                             review rating   fresh  \
0   3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1   3  It's an allegory in search of a meaning that n...    NaN  rotten   
2   3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3   3  Continuing along a line introduced in last yea...    NaN   fresh   
4   3             ... a perverse twist on neorealism...     NaN   fresh   

           critic  top_critic         publisher               date  
0      PJ Nabarro           0   Patrick Nabarro  November 10, 2018  
1  Annalee Newitz           0           io9.com       May 23, 2018  
2    Sean Axmaker           0  Stream on Demand    January 4, 2018  
3   Daniel Kasman           0              MUBI  November 16, 2017  
4             NaN           0      Cinema Scope   October 12, 2017

In [19]:
rt_reviews.shape

(54432, 8)

In [20]:
rt_reviews.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 54432 entries, 0 to 54431
Data columns (total 8 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   id          54432 non-null  int64 
 1   review      48869 non-null  object
 2   rating      40915 non-null  object
 3   fresh       54432 non-null  object
 4   critic      51710 non-null  object
 5   top_critic  54432 non-null  int64 
 6   publisher   54123 non-null  object
 7   date        54432 non-null  object
dtypes: int64(2), object(6)
memory usage: 3.3+ MB


The Rotten Tomatoes reviews dataset includes 54432 records where each record represents a movie.

The columns include id, review, rating, fresh, critic, top critic, publisher, and date.

A better measure of success is box office payout so ratings won't be a specific target metric we will focus on.

## THE MOVIE DB

In [29]:
tmdb_movies = pd.read_csv('/Users/nechamaborisute/Desktop/phase-2-project/zippedData/tmdb.movies.csv.gz', compression = 'gzip', index_col  = 0 ) 

In [30]:
tmdb_movies.head()

genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   
2        [12, 28, 878]  10138                en   
3      [16, 35, 10751]    862                en   
4        [28, 878, 12]  27205                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   
3                                     Toy Story      28.005   1995-11-22   
4                                     Inception      27.920   2010-07-16   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368  
3                                     Toy Story           7.9       10174  
4                                     Inception           8.3       22186

In [31]:
tmdb_movies.shape

(26517, 9)

In [32]:
tmdb_movies.info() # no nulls

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26517 entries, 0 to 26516
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   genre_ids          26517 non-null  object 
 1   id                 26517 non-null  int64  
 2   original_language  26517 non-null  object 
 3   original_title     26517 non-null  object 
 4   popularity         26517 non-null  float64
 5   release_date       26517 non-null  object 
 6   title              26517 non-null  object 
 7   vote_average       26517 non-null  float64
 8   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 2.0+ MB


The Movie DB dataset includes 26,517 records where each represents a movie. 

There are columns of id, genre ids, original language, original title, popularity, release date, title, vote average and vote count. 

There are no nulls in this dataset

## THE NUMBERS

In [25]:
tn_movie_budgets = pd.read_csv('/Users/nechamaborisute/Desktop/phase-2-project/zippedData/tn.movie_budgets.csv.gz', compression = 'gzip') 

In [26]:
tn_movie_budgets.head()

id  release_date                                        movie  \
0   1  Dec 18, 2009                                       Avatar   
1   2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2   3   Jun 7, 2019                                 Dark Phoenix   
3   4   May 1, 2015                      Avengers: Age of Ultron   
4   5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   

  production_budget domestic_gross worldwide_gross  
0      $425,000,000   $760,507,625  $2,776,345,279  
1      $410,600,000   $241,063,875  $1,045,663,875  
2      $350,000,000    $42,762,350    $149,762,350  
3      $330,600,000   $459,005,868  $1,403,013,963  
4      $317,000,000   $620,181,382  $1,316,721,747

In [27]:
tn_movie_budgets.shape

(5782, 6)

In [28]:
tn_movie_budgets.info() # no nulls

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


In [95]:
tn_movie_budgets.describe()

id  production_budget  domestic_gross  worldwide_gross  \
count  5782.000000       5.782000e+03    5.782000e+03     5.782000e+03   
mean     50.372363       3.158776e+07    4.187333e+07     9.148746e+07   
std      28.821076       4.181208e+07    6.824060e+07     1.747200e+08   
min       1.000000       1.100000e+03    0.000000e+00     0.000000e+00   
25%      25.000000       5.000000e+06    1.429534e+06     4.125415e+06   
50%      50.000000       1.700000e+07    1.722594e+07     2.798445e+07   
75%      75.000000       4.000000e+07    5.234866e+07     9.764584e+07   
max     100.000000       4.250000e+08    9.366622e+08     2.776345e+09   

       total_profit  
count  5.782000e+03  
mean   5.989970e+07  
std    1.460889e+08  
min   -2.002376e+08  
25%   -2.189071e+06  
50%    8.550286e+06  
75%    6.096850e+07  
max    2.351345e+09

The Numbers dataset includes 5782 records where each record represents a movie. 

The columns include id, release_date, movie, production_budget, domestic_gross and worldwide_gross. 

We can utilize the production budget and gross columns to determine profit.

## INTERNET MOVIE DATABASE: IMDB

In [47]:
conn = sqlite3.connect('/Users/nechamaborisute/Desktop/phase-2-project/zippedData/im.db')

In [49]:
%%bash
sqlite3 /Users/nechamaborisute/Desktop/phase-2-project/zippedData/im.db

.tables

directors      movie_akas     movie_ratings  principals   
known_for      movie_basics   persons        writers      


In [50]:
schema = pd.read_sql("""

SELECT *
FROM sqlite_schema 

""", conn)

In [51]:
schema

type           name       tbl_name  rootpage  \
0  table   movie_basics   movie_basics         2   
1  table      directors      directors         3   
2  table      known_for      known_for         4   
3  table     movie_akas     movie_akas         5   
4  table  movie_ratings  movie_ratings         6   
5  table        persons        persons         7   
6  table     principals     principals         8   
7  table        writers        writers         9   

                                                 sql  
0  CREATE TABLE "movie_basics" (\n"movie_id" TEXT...  
1  CREATE TABLE "directors" (\n"movie_id" TEXT,\n...  
2  CREATE TABLE "known_for" (\n"person_id" TEXT,\...  
3  CREATE TABLE "movie_akas" (\n"movie_id" TEXT,\...  
4  CREATE TABLE "movie_ratings" (\n"movie_id" TEX...  
5  CREATE TABLE "persons" (\n"person_id" TEXT,\n ...  
6  CREATE TABLE "principals" (\n"movie_id" TEXT,\...  
7  CREATE TABLE "writers" (\n"movie_id" TEXT,\n  ...

In [67]:
# table names
pd.read_sql("""SELECT name FROM sqlite_master WHERE type = 'table';""", conn)

name
0   movie_basics
1      directors
2      known_for
3     movie_akas
4  movie_ratings
5        persons
6     principals
7        writers

In [56]:
pd.read_sql("""

SELECT * 
FROM movie_basics
LIMIT 5

""", conn)

movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [57]:
pd.read_sql("""

SELECT COUNT(*) AS gnere_null_count
FROM movie_basics
WHERE genres IS NULL

""", conn)

gnere_null_count
0              5408

- columns include movie id, primary title, original title, start year, runtime minutes and genres.
> might drop these rows where genre is null since we need the genre unless we could combine with other dataframes to get the genre
> runtime minutes we could recommend how long the movie should be

In [55]:
pd.read_sql("""

SELECT *
FROM directors
LIMIT 5

""", conn)

movie_id  person_id
0  tt0285252  nm0899854
1  tt0462036  nm1940585
2  tt0835418  nm0151540
3  tt0835418  nm0151540
4  tt0878654  nm0089502

In [58]:
pd.read_sql("""

SELECT COUNT(*) AS person_id_null_count
FROM directors
WHERE person_id IS NULL

""", conn)

person_id_null_count
0                     0

- the director table includes movie id and person id
- no nulls in director table

In [59]:
pd.read_sql("""

SELECT * 
FROM known_for

""", conn).head()

person_id   movie_id
0  nm0061671  tt0837562
1  nm0061671  tt2398241
2  nm0061671  tt0844471
3  nm0061671  tt0118553
4  nm0061865  tt0896534

In [62]:
pd.read_sql("""

SELECT COUNT(*) AS movie_id_null_count
FROM known_for
WHERE movie_id IS NULL

""", conn).head()

movie_id_null_count
0                    0

In [69]:
pd.read_sql("""

SELECT COUNT(*) AS person_id_null_count
FROM known_for
WHERE person_id IS NULL

""", conn).head()

person_id_null_count
0                     0

- in known for table there are no nulls and it has  a person id in one column paired with the movie id of the movie their most known for

In [71]:
pd.read_sql("""

SELECT * 
FROM movie_akas
LIMIT 5

""", conn).head()

movie_id  ordering                                    title region  \
0  tt0369610        10                            Джурасик свят     BG   
1  tt0369610        11                        Jurashikku warudo     JP   
2  tt0369610        12  Jurassic World: O Mundo dos Dinossauros     BR   
3  tt0369610        13                  O Mundo dos Dinossauros     BR   
4  tt0369610        14                           Jurassic World     FR   

  language        types   attributes  is_original_title  
0       bg         None         None                0.0  
1     None  imdbDisplay         None                0.0  
2     None  imdbDisplay         None                0.0  
3     None         None  short title                0.0  
4     None  imdbDisplay         None                0.0

In [75]:
pd.read_sql("""

SELECT COUNT(*) types_null_count
FROM movie_akas
WHERE types IS NULL

""", conn).head()

types_null_count
0            163256

In [74]:
pd.read_sql("""

SELECT COUNT(*) attrib_null_count
FROM movie_akas
WHERE attributes IS NULL

""", conn).head()

attrib_null_count
0             316778

- the movie aka table contiains movie id, ordering, title, region, language, types, attributes, is original title 
- has bunch of nulls 

In [76]:
pd.read_sql("""

SELECT * 
FROM movie_ratings

""", conn).head()

movie_id  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21

In [77]:
pd.read_sql("""

SELECT COUNT(*) AS avgrating_null_count
FROM movie_ratings
WHERE averagerating IS NULL

""", conn).head()

avgrating_null_count
0                     0

In [78]:
pd.read_sql("""

SELECT COUNT(*) AS numvotes_null_count
FROM movie_ratings
WHERE numvotes IS NULL

""", conn).head()

numvotes_null_count
0                    0

In [79]:
pd.read_sql("""

SELECT COUNT(*) AS id_null_count
FROM movie_ratings
WHERE movie_id IS NULL

""", conn).head()

id_null_count
0              0

- the movie ratings table contains movie id, average rating and num votes
- no nulls

In [80]:
pd.read_sql("""

SELECT * 
FROM persons
LIMIT 5

""", conn)

person_id       primary_name birth_year death_year  \
0  nm0061671  Mary Ellen Bauder       None       None   
1  nm0061865       Joseph Bauer       None       None   
2  nm0062070         Bruce Baum       None       None   
3  nm0062195       Axel Baumann       None       None   
4  nm0062798        Pete Baxter       None       None   

                                 primary_profession  
0         miscellaneous,production_manager,producer  
1        composer,music_department,sound_department  
2                        miscellaneous,actor,writer  
3  camera_department,cinematographer,art_department  
4  production_designer,art_department,set_decorator

- the persons table contains person id, primary name, birth year, death year and primary profession.
- lots of nulls in birth and death columns 

In [81]:
pd.read_sql("""

SELECT * 
FROM principals
LIMIT 5

""", conn)

movie_id  ordering  person_id  category       job        characters
0  tt0111414         1  nm0246005     actor      None       ["The Man"]
1  tt0111414         2  nm0398271  director      None              None
2  tt0111414         3  nm3739909  producer  producer              None
3  tt0323808        10  nm0059247    editor      None              None
4  tt0323808         1  nm3579312   actress      None  ["Beth Boothby"]

- the principals table contains movie id, ordering, person id, category, job and characters.
- lots of nulls in job and characters columns 

In [84]:
pd.read_sql("""

SELECT *
FROM writers AS w
LIMIT 5 

""", conn)

movie_id  person_id
0  tt0285252  nm0899854
1  tt0438973  nm0175726
2  tt0438973  nm1802864
3  tt0462036  nm1940585
4  tt0835418  nm0310087

In [85]:
pd.read_sql("""

SELECT COUNT (*) writer_null_count
FROM writers AS w
WHERE person_id IS NULL

""", conn)

writer_null_count
0                  0

In [86]:
pd.read_sql("""

SELECT COUNT (*) movie_id_null_count
FROM writers AS w
WHERE movie_id IS NULL

""", conn)

movie_id_null_count
0                    0

- writers table contains movie id and person id 
- no nulls


The IMDB database is a relatively clean database with tables movie_basics, directors, known_for, movie_akas, movie_ratings, persons, principals and writers.


# Summary

As we can see, some these datasets include very useful information that we can use to draw helpful insights from, yet some are beyond the scope of the metrics we wish to focus on. For that reason, in the next notebook we will be analyzing and visualizing the data and drawing subsequent conclusions from the relevant datasets.